<?xml version="1.0"?>
<!--XML-->
    <launch>

    <arg name="gui" default="False" />
    <param name="use_gui" value="$(arg gui)"/>
    <param name="use_sim_time" value="true"/>

    <node name="load_pcl_list" pkg="pcl_tools" type="load_pcl_list" output="screen" >
        <param name="pcl2_output" type="string" value="/loaded_pcl2" />
        <param name="folderpath" type="string" value="$(find pcl_tools)/doc/examples/load_list/" />
        <param name="filename" type="string" value="test" />
        <param name="tf_extension" type="string" value="pose" />
        <param name="pcl_extension" type="string" value="ply" />
        <param name="point_cloud_frame_id" type="string" value="/camera" />
        <param name="world_frame_id" type="string" value="/world" />
        <param name="rate" type="double" value="5" />
        <param name="loop" type="bool" value="true" />
    </node>

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="0.01" />

        <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
        <param name="frame_id" type="string" value="/camera" />

        <!-- maximum range to integrate (speedup!) -->
        <param name="sensor_model/max_range" value="6" />

        <!-- data source to integrate (PointCloud2) -->
        <remap from="cloud_in" to="/loaded_pcl2" />
    </node>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pcl_tools)/rviz/octomap.rviz" />

</launch>
